process/elasticproc/logsevents/delegatorsProcessor.go

```diff
@@ -85,10 +90,26 @@ func (dp *delegatorsProc) processEvent(args *argsProcessEvent) argOutputProcessE
 		Contract:       contractAddr,
 		ActiveStake:    activeStake.String(),
 		ActiveStakeNum: dp.balanceConverter.ComputeBalanceAsFloat(activeStake),
+		Timestamp:      time.Duration(args.timestamp),
 	}
 
 	if eventIdentifierStr == withdrawFunc && len(topics) >= minNumTopicsDelegators+1 {
 		delegator.ShouldDelete = bytesToBool(topics[4])
+		withdrawFundIds := topics[5:]
+		delegator.WithdrawFundIDs = make([]string, 0)
+		for _, id := range withdrawFundIds {
+			delegator.WithdrawFundIDs = append(delegator.WithdrawFundIDs, hex.EncodeToString(id))
+		}
+	}
+	if eventIdentifierStr == unDelegateFunc && len(topics) >= minNumTopicsDelegators+1 {
```

Comment:
perhaps a new function that handles the UnDelegateInfo creation that takes the *data.Delegator parameter and does 2 early exits for the L104 cases?